llvm/lib/CodeGen/AsmPrinter/DwarfFile.h

```diff
@@ -124,6 +139,11 @@ class DwarfFile {
   /// Add a unit to the list of CUs.
   void addUnit(std::unique_ptr<DwarfCompileUnit> U);
 
+  /// Add a unit to the list of TUs.
+  /// Preserves type unit so that memory is not released before DWARF5
+  /// accelerator table is created.
```

Comment:
This comment doesn't seem quite accurate - it doesn't preserve any memory, right? It only saves an MCSymbol that can be used to reference a previously emitted type unit, right?